Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule_test.py

```diff
@@ -3166,6 +3168,338 @@ def __init__(self, status_code) -> None:
+
+    Args:
+        mocker (Any): The mocker object to patch the 'get_risk_score_user_or_host' method.
+        command (str): The command to be tested ('user' or 'host').
+        id (str): The ID parameter for the command.
+
+    Raises:
+        ValueError: If the expected exception is not raised or the error message doesn't match.
+
+    Returns:
+        None
+    """
+
+    client = CoreClient(
+        base_url="test",
+        headers={},
+    )
+
+    class MockException:
+        def __init__(self, status_code) -> None:
+            self.status_code = status_code
+
+    mocker.patch.object(
+        client,
+        "get_risk_score_user_or_host",
+        side_effect=DemistoException(
+            message="id 'test' was not found", res=MockException(500)
+        ),
+    )
+    with pytest.raises(
+        ValueError,
+        match="Error: id test was not found. Full error message: id 'test' was not found"
+    ):
+        get_list_risky_users_or_host_command(client, command, {id: "test"})
+
+
+def test_get_list_user_groups_command(mocker):
+    """
+    Test function to validate the behavior of the `get_list_user_groups_command` function.
+
+    Args:
+        mocker: Pytest mocker object.
+        args (dict): A dictionary containing optional `group_names` argument.
+
+    Returns:
+        None
+
+    Raises:
+        AssertionError: If the expected output doesn't match the actual output.
+    """
+    client = CoreClient("test", {})
+    test_data = load_test_data("./test_data/get_list_user_groups.json")
+    mocker.patch.object(CoreClient, "get_list_user_groups", return_value=test_data)
+
+    results = get_list_user_groups_command(client=client, args={"group_names": "test"})
+    assert test_data["reply"] == results.outputs
+
+
+@pytest.mark.parametrize(
+    "data, expected_results",
+    [
+        (
+            {
+                "group_name": "Group2",
+                "description": None,
+                "pretty_name": "dummy1",
+                "insert_time": 1111111111111,
+                "update_time": 2222222222222,
+                "user_email": [
+                    "dummy1@gmail.com",
+                    "dummy2@gmail.com",
+                ],
+                "source": "Custom",
+            },
+            [
+                {'User email': 'dummy1@gmail.com', 'Group Name': 'Group2', 'Group Description': None},
+                {'User email': 'dummy2@gmail.com', 'Group Name': 'Group2', 'Group Description': None}
+            ]
+        )
+    ],
+)
+def test_parse_user_groups(data: dict[str, Any], expected_results: list[dict[str, Any]]):
+    """
+    Test the 'parse_user_groups' function that parses user group information.
+
+    Args:
+        data (dict): A dictionary containing a sample user group data.
+
+    Returns:
+        None
+
+    Raises:
+        AssertionError: If the parsing of user groups data fails.
+    """
+    results = parse_user_groups(data)
+
+    assert expected_results == results
```

Comment:
```suggestion
    assert parse_user_groups(data) == expected_results
```
1. It's convention to use the expected on the right-hand side of the equation
2. there's no real for a variable